waspc/src/Wasp/Generator/ServerGenerator.hs

```diff
@@ -91,33 +93,34 @@ genDotEnv spec =
 dotEnvInServerRootDir :: Path' (Rel C.ServerRootDir) File'
 dotEnvInServerRootDir = [relfile|.env|]
 
-genReadme :: FileDraft
-genReadme = C.mkTmplFd (asTmplFile [relfile|README.md|])
-
-genPackageJson :: AppSpec -> [AS.Dependency.Dependency] -> [AS.Dependency.Dependency] -> FileDraft
-genPackageJson spec waspDeps waspDevDeps =
-  C.mkTmplFdWithDstAndData
-    (asTmplFile [relfile|package.json|])
-    (asServerFile [relfile|package.json|])
-    ( Just $
-        object
-          [ "depsChunk" .= npmDepsToPackageJsonEntry (resolvedWaspDeps ++ resolvedUserDeps),
-            "devDepsChunk" .= npmDevDepsToPackageJsonEntry waspDevDeps,
-            "nodeVersion" .= nodeVersionAsText,
-            "startProductionScript"
-              .= if not (null $ AS.getDecls @AS.Entity.Entity spec)
-                then "npm run db-migrate-prod && "
-                else
-                  ""
-                    ++ "NODE_ENV=production node ./src/server.js"
-          ]
-    )
-  where
-    (resolvedWaspDeps, resolvedUserDeps) =
-      case resolveNpmDeps waspDeps userDeps of
-        Right deps -> deps
-        Left depsAndErrors -> error $ intercalate " ; " $ map snd depsAndErrors
+genReadme :: Generator FileDraft
+genReadme = return $ C.mkTmplFd (asTmplFile [relfile|README.md|])
+
+genPackageJson :: AppSpec -> [AS.Dependency.Dependency] -> [AS.Dependency.Dependency] -> Generator FileDraft
+genPackageJson spec waspDeps waspDevDeps = do
+  (resolvedWaspDeps, resolvedUserDeps) <-
+    case resolveNpmDeps waspDeps userDeps of
+      Right deps -> return deps
+      Left depsAndErrors -> logAndThrowGeneratorError $ GenericGeneratorError $ intercalate " ; " $ map snd depsAndErrors
 
+  return $
+    C.mkTmplFdWithDstAndData
+      (asTmplFile [relfile|package.json|])
+      (asServerFile [relfile|package.json|])
+      ( Just $
+          object
+            [ "depsChunk" .= npmDepsToPackageJsonEntry (resolvedWaspDeps ++ resolvedUserDeps),
+              "devDepsChunk" .= npmDevDepsToPackageJsonEntry waspDevDeps,
+              "nodeVersion" .= nodeVersionAsText,
+              "startProductionScript"
+                .= if not (null $ AS.getDecls @AS.Entity.Entity spec)
+                  then "npm run db-migrate-prod && "
+                  else
+                    ""
+                      ++ "NODE_ENV=production node ./src/server.js"
```

Comment:
I know this is old code but this is weird :D, maybe we can remove this `"" ++` part as it does nothing, I think?

Reply:
Ah actually no, it is not nothing, that "" is part of `else`!
Precedence is like this I think:
```hs
(if not (null $ AS.getDecls @AS.Entity.Entity spec)
 then "npm run db-migrate-prod && "
 else ""
) ++ "NODE_ENV=production node ./src/server.js"
```

Since we have an opportunity here, would you mind somehow making this piece of code more clear, by using parentheses or maybe fixing indentation?

Reply:
This actually may be a bug you found. As is, the server's package.json has the following:

```
    "start-production": "npm run db-migrate-prod && ",
```

It does not even include the NODE_ENV stuff!

If I update it to:
```haskell
              "startProductionScript"
                .= ( ( if not (null $ AS.getDecls @AS.Entity.Entity spec)
                         then "npm run db-migrate-prod && "
                         else ""
                     )
                       ++ "NODE_ENV=production node ./src/server.js"
                   )
```

We now get:
```
    "start-production": "npm run db-migrate-prod && NODE_ENV=production node ./src/server.js",
```

Good spot on style that turned out to be bigger!! 👍🏻 Seems it was introduced during lint cleanup: https://github.com/wasp-lang/wasp/commit/89c9d6a1dbc9088413aee6da2d1e5ea9b8275010

Reply:
We could even go back to the old syntax as it may be cleaner:
```haskell
            .= concat
                [ if not (null $ Wasp.getPSLEntities wasp) then "npm run db-migrate-prod && " else "",
                  "NODE_ENV=production node ./src/server.js"
                ]
```

Reply:
Ah uff, thanks a lot for taking the time to check that out!

Yeah sounds good (the last option)!

Reply:
I was able to keep linter happy with this (doesn't like concat on 2 elements) 👍🏻 :

```haskell
                .= ( (if not (null $ AS.getDecls @AS.Entity.Entity spec) then "npm run db-migrate-prod && " else "")
                       ++ "NODE_ENV=production node ./src/server.js"
                   )
```